tests/Type/AbstractSentryTest.php

```diff
@@ -215,7 +226,31 @@ public function setFoo($newValue)
 	{
 		$this->fooProperty = $newValue;
 	}';
-		Assert::assertSame($method, $sentry->generateMethod($propertyMetadata, $setMethod));
+
+			return [
+				'propertyMetadata' => $propertyMetadata,
+				'sentryMethod' => $sentryMethod,
+				'expectedMethod' => $expectedMethod,
+			];
+		})();
+	}
+
+	/**
+	 * @dataProvider generateMethodDataProvider
+	 *
+	 * @param \Consistence\Sentry\Metadata\PropertyMetadata $propertyMetadata
+	 * @param \Consistence\Sentry\Metadata\SentryMethod $sentryMethod
+	 * @param string $expectedMethod
+	 */
+	public function testGenerateMethod(
+		PropertyMetadata $propertyMetadata,
+		SentryMethod $sentryMethod,
+		string $expectedMethod
```

Comment:
I think `expectedGeneratedMethod` would be more descriptive?

Reply:
Yes, renamed.